File: schedule.py
from datetime import datetime

file = open("C:/Users/agupta-22/Magic-Mirror/schedule.asd", "w")

time = datetime.now().strftime("%H:%M")
time2 = datetime.now().strftime("%I:%M")
headings = []
headings = time2
headings = headings.replace("01:", "1:")
headings = headings.replace("02:", "2:")
headings = headings.replace("03:", "3:")
headings = headings.replace("04:", "4:")
headings = headings.replace("05:", "5:")
headings = headings.replace("06:", "6:")
headings = headings.replace("07:", "7:")
headings = headings.replace("08:", "8:")
headings = headings.replace("09:", "9:")
if time < "12:00":
    file.write(headings + " AM \n")
    
elif time >= "12:00":
    file.write(headings + " PM \n")
weekday = datetime.today().strftime("%A")
print (time)
# file.write("Current Block: ")    
# if time >= "8:00" and time <= "8:50" and weekday == "Monday":
#     file.write("A")
# elif time >= "9:00" and time <= "10:15" and weekday == "Monday":
#     file.write("B")
# elif time >= "10:25" and time <= "11:25" and weekday == "Monday":
#     file.write("C")
# elif time >= "11:25" and time <= "12:15" and weekday == "Monday":
#     file.write("Chapel")
# elif time > "12:15" and time <= "12:55" and weekday == "Monday":
#     file.write("Lunch")
# elif time > "12:55" and time <= "14:10" and weekday == "Monday":
#     file.write("D")
# elif time >= "14:20" and time <= "15:10" and weekday == "Monday":
#     file.write("E")

# if time >= "8:00" and time <= "8:50" and weekday == "Tuesday":
#     file.write("F")
# elif time >= "9:00" and time <= "10:15" and weekday == "Tuesday":
#     file.write("G")
# elif time >= "10:25" and time <= "11:25" and weekday == "Tuesday":
#     file.write("A")
# elif time >= "11:25" and time <= "12:15" and weekday == "Tuesday":
#     file.write("DMX")
# elif time > "12:15" and time <= "13:20" and weekday == "Tuesday":
#     file.write("Lunch")
# elif time > "13:20" and time <= "14:10" and weekday == "Tuesday":
#     file.write("C")
# elif time >= "14:20" and time <= "15:10" and weekday == "Tuesday":
#     file.write("B")

# if time >= "8:00" and time <= "8:50" and weekday == "Wednesday":
#     file.write("D")
# elif time >= "9:00" and time <= "10:15" and weekday == "Wednesday":
#     file.write("E")
# elif time >= "10:25" and time <= "11:25" and weekday == "Wednesday":
#     file.write("G")
# elif time >= "11:25" and time <= "12:15" and weekday == "Wednesday":
#     file.write("Community Meeting")
# elif time > "12:15" and time < "12:55" and weekday == "Wednesday":
#     file.write("Lunch")
# elif time >= "12:55" and time <= "13:45" and weekday == "Wednesday":
#     file.write("F")

# if time >= "8:00" and time <= "8:50" and weekday == "Thursday":
#     file.write("B")
# elif time >= "9:00" and time <= "10:15" and weekday == "Thursday":
#     file.write("C")
# elif time >= "10:25" and time <= "11:15" and weekday == "Thursday":
#     file.write("D")
# elif time >= "11:25" and time <= "12:15" and weekday == "Thursday":
#     file.write("DMX")
# elif time > "12:15" and time <= "13:20" and weekday == "Thursday":
#     file.write("Lunch")
# elif time > "13:20" and time <= "14:10" and weekday == "Thursday":
#     file.write("E")
# elif time >= "14:20" and time <= "15:10" and weekday == "Thursday":
#     file.write("A")

# if time >= "8:00" and time <= "8:50" and weekday == "Friday":
#     file.write("G")
# elif time >= "9:00" and time <= "10:15" and weekday == "Friday":
#     file.write("F")
# elif time >= "10:25" and time <= "11:25" and weekday == "Friday":
#     file.write("B")
# elif time >= "11:25" and time <= "12:15" and weekday == "Friday":
#     file.write("Chapel")
# elif time > "12:15" and time <= "12:55" and weekday == "Friday":
#     file.write("Lunch")
# elif time > "12:55" and time <= "14:10" and weekday == "Friday":
#     file.write("A")
# elif time >= "14:20" and time <= "15:10" and weekday == "Friday":
#     file.write("C")

# if time >= "8:00" and time <= "8:50" and weekday == "Saturday":
#     file.write("E")
# elif time >= "9:00" and time <= "9:50" and weekday == "Saturday":
#     file.write("D")
# elif time >= "10:00" and time <= "10:50" and weekday == "Saturday":
#     file.write("F")
# elif time >= "11:00" and time <= "11:50" and weekday == "Saturday":
#     file.write("G")

# file.close()

File: schedule.js
var day = new Date();
var today = parseInt(day.getDay());
var hour = parseInt(day.getHours());
var minutes = parseInt(day.getMinutes());
var imgElem = document.getElementById('image_1');

(function () {
  if (today == 1) {
    if (hour == 8 && minutes <= 50) { // Monday A Block
      imgElem.src = 'MonA.PNG';
    }
    else if (hour == 9) { // Monday B Block
      imgElem.src = 'MonB.PNG';
    }
    else if (hour == 10 && minutes >= 0 && minutes <= 15) { // Monday B Block
      imgElem.src = 'MonB.PNG';
    }
    else if (hour == 10 && minutes >= 20) { // Monday Chapel and Conference Block
      imgElem.src = 'MonChapel.PNG';
    }
    else if (hour == 11 && minutes >= 0 && minutes <= 15) { // Monday Chapel and Conference Block
      imgElem.src = 'MonChapel.PNG';
    }
    else if (hour == 11 && minutes >=20) { // Monday C Block
      imgElem.src = 'MonC.PNG';
    }
    else if (hour == 12 && minutes <=10) { // Monday C Block
      imgElem.src = 'MonC.PNG';
    }
    else if (hour == 12 && minutes > 10 && minutes < 55) { // Monday Lunch
      imgElem.src = 'MonLunch.PNG';
    }
    else if (hour == 12 && minutes >=55) { // Monday D Block
      imgElem.src = 'MonD.PNG';
    }
    else if (hour == 13) { // Monday D Block
      imgElem.src = 'MonD.PNG';
    }
    else if (hour == 14 && minutes >= 0 && minutes < 10) { // D Block
      imgElem.src = 'MonD.PNG';
    }
    else if (hour == 14 && minutes >= 10) { // Monday E Block
      imgElem.src = 'MonE.PNG';
    }
    else if (hour == 15 && minutes >= 0 && minutes < 10) { // E Block
      imgElem.src = 'MonE.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 2) {
    if (hour == 8 && minutes <= 50) { // Tuesday F Block
      imgElem.src = 'TuesF.PNG';
    }
    else if (hour == 9) { // Tuesday G Block
      imgElem.src = 'TuesG.PNG';
    }
    else if (hour == 10 && minutes >= 0 && minutes <= 15) { // Tuesday G Block
      imgElem.src = 'TuesG.PNG';
    }
    else if (hour == 10 && minutes >= 20) { // Tuesday DMX Block
      imgElem.src = 'TuesDMX.PNG';
    }
    else if (hour == 11 && minutes >= 0 && minutes <= 15) { // Tuesday DMX Block
      imgElem.src = 'TuesDMX.PNG';
    }
    else if (hour == 11 && minutes >=15) { // Tuesday A Block
      imgElem.src = 'TuesA.PNG';
    }
    else if (hour == 12 && minutes <=10) { // Tuesday A BLock
      imgElem.src = 'TuesA.PNG';
    }
    else if (hour == 12 && minutes >10) { // Tuesday Lunch 
      imgElem.src = 'TuesLunch.PNG';
    }
    else if (hour == 13 && minutes >= 0 && minutes < 20) { // Tuesday Lunch 
      imgElem.src = 'TuesLunch.PNG';
    }
    else if (hour == 13 && minutes >=20) { // Tuesday C Block
      imgElem.src = 'TuesC.PNG';
    }
    else if (hour == 14 && minutes >= 0 && minutes < 10) { // Tuesday C Block
      imgElem.src = 'TuesC.PNG';
    }
    else if (hour == 14 && minutes >= 10) { // Tuesday B Block
      imgElem.src = 'TuesB.PNG';
    }
    else if (hour == 15 && minutes >= 0 && minutes < 10) { // Tuesday B Block
      imgElem.src = 'TuesB.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 3) {
    if (hour == 8 && minutes <= 50) { // Wednesday D Block
      imgElem.src = 'WedD.PNG';
    }
    else if (hour == 9) { // Wednesday E Block
      imgElem.src = 'WedE.PNG';
    }
    else if (hour == 10 && minutes >= 20) { // Wednesday Community Meeting
      imgElem.src = 'WedComm.PNG';
    }
    else if (hour == 11 && minutes >= 0 && minutes <= 15) { // Wednesday Community Meeting
      imgElem.src = 'WedComm.PNG';
    }
    else if (hour == 11 && minutes >=20) { // Wednesday G Block
      imgElem.src = 'WedG.PNG';
    }
    else if (hour == 12 && minutes <= 10) { // Wednesday G Block
      imgElem.src = 'WedG.PNG';
    }
    else if (hour == 12 && minutes > 10 && minutes<55) { // Wednesday Lunch
      imgElem.src = 'WedLunch.PNG';
    }
    else if (hour == 12 && minutes>=55) { // Wednesday F Block
      imgElem.src = 'WedF.PNG';
    }
    else if (hour == 13 && minutes >= 0 && minutes <= 45 ) { // Wednesday F Block
      imgElem.src = 'WedF.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 4) {
    if (hour == 8 && minutes <= 50) { // Thursday B Block
      imgElem.src = 'ThursB.PNG';
    }
    else if (hour == 9) { // Thursday C Block
      imgElem.src = 'ThursC.PNG';
    }
    else if (hour == 10 && minutes >= 0 && minutes <= 15) { // Thursday C Block
      imgElem.src = 'ThursC.PNG';
    }
    else if (hour == 10 && minutes >= 20) { // Thursday DMX Block
      imgElem.src = 'ThursDMX.PNG';
    }
    else if (hour == 11 && minutes >= 0 && minutes <= 15) { // Thursday DMX Block
      imgElem.src = 'ThursDMX.PNG';
    }
    else if (hour == 11 && minutes >=15) { // Thursday D Block
      imgElem.src = 'ThursD.PNG';
    }
    else if (hour == 12 && minutes <=10) { // Thursday D BLock
      imgElem.src = 'ThursD.PNG';
    }
    else if (hour == 12 && minutes >10) { // Thursday Lunch 
      imgElem.src = 'ThursLunch.PNG';
    }
    else if (hour == 13 && minutes >= 0 && minutes < 20) { // Thursday Lunch 
      imgElem.src = 'ThursLunch.PNG';
    }
    else if (hour == 13 && minutes >=20) { // Thursday E Block
      imgElem.src = 'ThursE.PNG';
    }
    else if (hour == 14 && minutes >= 0 && minutes < 10) { // Thursday E Block
      imgElem.src = 'ThursE.PNG';
    }
    else if (hour == 14 && minutes >= 10) { // Thursday A Block
      imgElem.src = 'ThursA.PNG';
    }
    else if (hour == 15 && minutes >= 0 && minutes < 10) { // Thursday A Block
      imgElem.src = 'ThursA.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 5) {
    if (hour == 8 && minutes <= 50) { // Friday G Block
      imgElem.src = 'FriG.PNG';
    }
    else if (hour == 9) { // Friday F Block
      imgElem.src = 'FriF.PNG';
    }
    else if (hour == 10 && minutes >= 0 && minutes <= 15) { // Friday F Block
      imgElem.src = 'FriF.PNG';
    }
    else if (hour == 10 && minutes >= 20) { // Friday Chapel and Conference Block
      imgElem.src = 'FriChapel.PNG';
    }
    else if (hour == 11 && minutes >= 0 && minutes <= 15) { // Friday Chapel and Conference Block
      imgElem.src = 'FriChapel.PNG';
    }
    else if (hour == 11 && minutes >=20) { // Friday B Block
      imgElem.src = 'FriB.PNG';
    }
    else if (hour == 12 && minutes <=10) { // Friday C Block
      imgElem.src = 'FriB.PNG';
    }
    else if (hour == 12 && minutes > 10 && minutes < 55) { // Friday Lunch
      imgElem.src = 'FriLunch.PNG';
    }
    else if (hour == 12 && minutes >=55) { // Friday A Block
      imgElem.src = 'FriA.PNG';
    }
    else if (hour == 13) { // Friday A Block
      imgElem.src = 'FriA.PNG';
    }
    else if (hour == 14 && minutes >= 0 && minutes < 10) { // Friday A Block
      imgElem.src = 'FriA.PNG';
    }
    else if (hour == 14 && minutes >= 10) { // Friday C Block
      imgElem.src = 'FriC.PNG';
    }
    else if (hour == 15 && minutes >= 0 && minutes < 10) { // Friday C Block
      imgElem.src = 'FriC.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 6) {
    if (hour == 8 && minutes <= 50) { // Saturday E Block
      imgElem.src = 'SatE.PNG';
    }
    else if (hour == 9 && minutes <= 50) { // Saturday D Block
      imgElem.src = 'SatD.PNG';
    }
    else if (hour == 10  && minutes <= 50) { // Saturday F Block
      imgElem.src = 'SatF.PNG';
    }
    else if (hour == 11 && minutes <= 50) { // Saturday G Block
      imgElem.src = 'SatG.PNG';
    }
    else {
      imgElem.src = 'DailySchedule.PNG';
    }
  }
  if (today == 0) {
    imgElem.src = 'DailySchedule.PNG';
  }
  if (hour == 8 && minutes == 0 && seconds ==0 ) {
    window.location.reload();
  }
  else if (hour == 9 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 10 && minutes == 20 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 11 && minutes == 20 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 12 && minutes == 10 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 14 && minutes == 20 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 15 && minutes > 10 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 16 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 17 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 18 && minutes == 0 && seconds ==0  ) { 
    window.location.reload();
  }
  else if (hour == 19 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 20 && minutes == 0 && seconds ==0)  { 
    window.location.reload();
  }
  else if (hour == 21 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 22 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 23 && minutes == 0 && seconds ==0) { 
    window.location.reload();
  }
  else if (hour == 24 && minutes ==0 && seconds ==0) { 
    window.location.reload();
  }
  if (today == 1 || today == 3 || today ==5) {
    if (hour == 12 && minutes == 55 && seconds ==0) {
      window.location.reload();
    }
  }
  else if (today == 2 || today == 4) {
    if (hour == 13 && minutes == 20 && seconds ==0) {
      window.location.reload();
  }
  }
  else if (today == 6 ) {
    if (hour == 11 && minutes == 51 && seconds ==0) {
      window.location.reload();
  }
  }
  else if (today == 3 ) {
    if (hour == 13 && minutes == 46 && seconds ==0) {
      window.location.reload();
  }
  else if (hour == 1  && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 2 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 3 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 4 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 5 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 6 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  else if (hour == 7 && minutes == 0 && seconds ==0 ) { 
    window.location.reload();
  }
  }

})();